hertzbeat-manager/src/main/java/org/apache/hertzbeat/manager/service/impl/MonitorServiceImpl.java

```diff
@@ -608,7 +608,7 @@ public Page<Monitor> getMonitors(List<Long> monitorIds, String app, String searc
             if (StringUtils.hasText(search)) {
                 Predicate predicateHost = criteriaBuilder.like(root.get("host"), "%" + search + "%");
                 Predicate predicateName = criteriaBuilder.like(criteriaBuilder.lower(root.get("name")), "%" + search.toLowerCase() + "%");
-                if (CommonUtil.isNumeric(search)){
+                if (CommonUtil.isLong(search)){
                     Predicate predicateId = criteriaBuilder.equal(root.get("id"), Long.parseLong(search));
```

Comment:
Long.parseLong can still fail if the number has lots of digits, eg 20 or more.
9,223,372,036,854,775,807 is the max long in Java.